Library/Homebrew/cmd/update.sh

```diff
@@ -9,31 +9,39 @@
 #:      -d, --debug                      Display a trace of all shell commands as they are executed.
 #:      -h, --help                       Show this message.
 
-source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
+# HOMEBREW_CURLRC, HOMEBREW_DEVELOPER, HOMEBREW_FAILED_FETCH_DIRS, HOMEBREW_GIT_EMAIL
```

Comment:
`HOMEBREW_FAILED_FETCH_DIRS` can be deleted entirely, it seems to never be set.

Reply:
Seems to have been added at the same time as `HOMEBREW_MISSING_REMOTE_REF_DIRS`, which ends up not being used. Perhaps just a mismatch of names here?

Reply:
Any thoughts @Rylan12? Yeh, I think `HOMEBREW_MISSING_REMOTE_REF_DIRS` is probably the right thing to use here instead.

Reply:
Yeah, I'm not totally sure what happened here. It looks like `HOMEBREW_FAILED_FETCH_DIRS` should be `HOMEBREW_MISSING_REMOTE_REF_DIRS` here and in `update-report`. I guess this hasn't been working this whole time...? That's kind of annoying...

I just did a quick test locally, though, and can confirm that the expected output is _not_ displayed by default but _is_ displayed by changing to use `HOMEBREW_MISSING_REMOTE_REF_DIRS`.

My guess is that I wanted to change the name from `HOMEBREW_FAILED_FETCH_DIRS` to `HOMEBREW_MISSING_REMOTE_REF_DIRS` before merging the PR and forgot to rename all the references and never re-tested 🤦

Reply:
@hyuraku can you change `HOMEBREW_FAILED_FETCH_DIRS` to `HOMEBREW_MISSING_REMOTE_REF_DIRS` both here and on [line 172 of `update-report`](https://github.com/Homebrew/brew/blob/9dadfc20c49d9be9fada4bb804595f3bd67a02b1/Library/Homebrew/cmd/update-report.rb#L172)?

Reply:
@Rylan12 
Sure, I will change `update.sh` and `update-report.sh`